cibyl/plugins/openstack/sources/jenkins_job_builder.py

```diff
@@ -32,25 +34,45 @@
 NODE_NAME_COUNTER = r'[a-zA-Z]+:[0-9]+'
 IP_VERSION = r'--network-protocol\s+ipv[4|6]'
 IP_VERSION_NUMBER = r'4|6'
+RELEASE = r'.*rhos-\d\d.\d-.*patches|.*rhos-\d\d-.*patches|.*send_results_to_umb.*'  # noqa: E501
+RELEASE_NUMBER = r'\d\d.\d|\d\d'
+CINDER_BACKEND = r'.*--storage-backend.*|.*IR_TRIPLEO_OVERCLOUD_STORAGE_BACKEND_UPD.*'  # noqa: E501
+CINDER_BACKEND_NAME = r'ceph|lvm|netapp-iscsi|netapp-nfs|swift|nfs'
+
+
+def args_are_in_list(arg_list, list):
+    """
+    Find if elements of 'arg_list' are memebers of 'list'
+
+    param: list arg_list: e.g. a list of strings from kwargs['release'].value
+    param: list list: e.g. a list of strings to match against
+    return: True/Flase
+    rtype: Bool
+    """
+    return(len([el for el in list if " ".join(map(str, arg_list)) in el]) > 0)
+
+
+@cache
```

Comment:
It seems that `cache` is not available in py36. We should move to `lru_cache` to support that version [docs](https://docs.python.org/3.6/library/functools.html#functools.lru_cache)

Reply:
Looks line caching 2500 config.yaml files is stretching the functools abilities and that why cibyl query  --release 16.2 --cinder-backend lvm returns 0 jobs. Removing it meanwhile and let's adopt serialization with pickle instead.